src/Map.js

```diff
@@ -169,39 +169,101 @@ export function Map(props){
           first++;
         }
+         *
+         */
+
+        // Find out whether there are any events in the list that are not private
+        let first = -1;  
+        for (let i=0; i<props.events.length; i++) {
+            if (
+              ('location' in props.events[i]) && 
+              ('location' in props.events[i]['location']) && 
+              ('latitude' in props.events[i]['location']['location'])
+            ) 
+            {
+              first = i;
+              break;
+            }
+        }
 
-        var lat = props.events[first]['location']['location']['latitude'];
-        var long = props.events[first]['location']['location']['longitude'];
+        if (first !== -1) {  // There is at least one event that is not private (hence visible on map)
 
-        if(center[0] !== lat || center[0] !== long){
-          setCenter([lat, long]);
-          setNewCenter(true);
-        }
+            var lat = props.events[first]['location']['location']['latitude'];
+            var long = props.events[first]['location']['location']['longitude'];
+
+            if(center[0] !== lat || center[0] !== long){
+              setCenter([lat, long]);
+              setNewCenter(true);
+            }
 
-        var places = {};
+            var places = {};
 
-        props.events.forEach(function(event, index) {
+            props.events.forEach(function(event, index) {
 
-          //If has longitude and latitute
-          if ('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']) {
+              //If has longitude and latitute
+              if ('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']) {
 
-            //Creates string key for {places} dictionary
-            let str = event['location']['location']['latitude'] + "&" + event['location']['location']['longitude'];
-            //Creates or adds to a location - adds HTML code for event list for that location
-            if (str in places) {
-              places[str] = places[str] + 1;
-            } else {
-              places[str] = 1;
-            }
+                //Creates string key for {places} dictionary
+                let str = event['location']['location']['latitude'] + "&" + event['location']['location']['longitude'];
+                //Creates or adds to a location - adds HTML code for event list for that location
+                if (str in places) {
+                  places[str] = places[str] + 1;
+                } else {
+                  places[str] = 1;
+                }
 
-          }
-        });
-        setLocations(places);
+              }
+            });
+            setLocations(places);
+        }
+        else {
+            /* 
+             * frm: TODO: After everyone is OK with the changes to account for 
```

Comment:
Seems reasonable

Reply:
Leaving this in the code for now as a TODO because the refactor will make it hard for someone to grok the diffs.  I will come back again later on to fix, but the code as-is is not too bad / hard to understand.